inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/RedisReader.java

```diff
@@ -0,0 +1,445 @@
+import com.google.gson.stream.JsonWriter;
+import com.moilioncircle.redis.replicator.RedisReplicator;
+import com.moilioncircle.redis.replicator.Replicator;
+import com.moilioncircle.redis.replicator.cmd.CommandName;
+import com.moilioncircle.redis.replicator.cmd.impl.DefaultCommand;
+import com.moilioncircle.redis.replicator.cmd.parser.DefaultCommandParser;
+import com.moilioncircle.redis.replicator.event.Event;
+import com.moilioncircle.redis.replicator.event.EventListener;
+import com.moilioncircle.redis.replicator.event.PostRdbSyncEvent;
+import com.moilioncircle.redis.replicator.rdb.datatype.KeyStringValueHash;
+import com.moilioncircle.redis.replicator.rdb.datatype.KeyStringValueList;
+import com.moilioncircle.redis.replicator.rdb.datatype.KeyStringValueSet;
+import com.moilioncircle.redis.replicator.rdb.datatype.KeyStringValueString;
+import com.moilioncircle.redis.replicator.rdb.datatype.KeyStringValueZSet;
+import com.moilioncircle.redis.replicator.rdb.datatype.KeyValuePair;
+import com.moilioncircle.redis.replicator.rdb.datatype.ZSetEntry;
+import org.apache.inlong.agent.conf.JobProfile;
+import org.apache.inlong.agent.message.DefaultMessage;
+import org.apache.inlong.agent.metrics.audit.AuditUtils;
+import org.apache.inlong.agent.plugin.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * Redis data reader
+ */
+public class RedisReader extends AbstractReader {
+
+    public static final String REDIS_READER_TAG_NAME = "AgentRedisMetric";
+    public static final String JOB_REDIS_PORT = "job.redisJob.port";
+    public static final String JOB_REDIS_HOSTNAME = "job.redisJob.hostname";
+    public static final String JOB_REDIS_SSL = "job.redisJob.ssl";
+    public static final String JOB_REDIS_AUTHUSER = "job.redisJob.authUser";
+    public static final String JOB_REDIS_AUTHPASSWORD = "job.redisJob.authPassword";
+    public static final String JOB_REDIS_READTIMEOUT = "job.redisJob.readTimeout";
+    public static final String JOB_REDIS_QUEUE_SIZE = "job.redisJob.queueSize";
+    public static final String JOB_REDIS_REPLID = "job.redisJob.replId";
+    public static final String JOB_REDIS_OFFSET = "job.redisJob.offset";
+    private static final Logger LOGGER = LoggerFactory.getLogger(RedisReader.class);
+    private String port;
+    private String hostName;
+    private boolean ssl;
+    private String authUser;
+    private String authPassword;
+    private String readTimeout;
+    private String instanceId;
+    private String replId;
+    private String snapShot;
+    private boolean destroyed;
+    private Replicator redisReplicator;
+    private LinkedBlockingQueue<String> redisMessageQueue;
+    private boolean finished = false;
+    private ExecutorService executor;
+    private Gson gson;
+
```

Comment:
Suggest to add filter

Reply:
what kind of filter

Reply:
https://redis.io/docs/manual/programmability/eval-intro/
@iamsee123 I would like you support scripting with Lua, thanks answer to the question.

Reply:
By using lua-replicate-commands configuration directive and [redis.replicate_commands()](https://redis.io/topics/lua-api#redis.replicate_commands) Lua API can enable effects replication, then redisReader can read the command directly. However if using verbatim replication, we just read the eval command but the change of data can be read by rdb scan at the beginning.

Reply:
OK